File: fix_file.py
file_path = 'c:/Users/Engineer Kairo Maina/Desktop/Kairo 4K-Streamer-/src/components/admin/VirtualChannelManager.tsx'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Fix the corrupted content - literal \n strings (backslash followed by n)
content = content.replace('                \\n', '        }')
content = content.replace('        \\n// Remove undefined values', '        // Remove undefined values')
content = content.replace('                \\n\\n        // Remove undefined values', '        }\\n\\n        // Remove undefined values')

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(content)

print('File fixed!')

File: fix_file.js
const fs = require('fs');
const path = 'c:/Users/Engineer Kairo Maina/Desktop/Kairo 4K-Streamer-/src/components/admin/VirtualChannelManager.tsx';
let content = fs.readFileSync(path, 'utf8');

// Fix the corrupted addToSchedule function
const oldPattern = /        \}        \/\/ Remove undefined values to avoid Supabase errors\\n        if \(payload\.media_id === undefined\) \{ delete payload\.media_id; \}\\n        if \(payload\.ad_id === undefined\) \{ delete payload\.ad_id; \}\\n\\n        const \{ error \} = await supabase\.from\('channel_schedule'\)\.insert\(\[payload\]\);/g;
const newCode = `        }

        // Remove undefined values to avoid Supabase errors
        if (payload.media_id === undefined) { delete payload.media_id; }
        if (payload.ad_id === undefined) { delete payload.ad_id; }

        const { error } = await supabase.from('channel_schedule').insert([payload]);`;

content = content.replace(oldPattern, newCode);
fs.writeFileSync(path, content);
console.log('File fixed!');
